juspay/hyperswitch
# fix(opensearch): show search results only if user has access permission to the index

## Type of Change



## Description

/search and /search/{domain} are currently checking for analytics ACL before hitting opensearch and returning results.

Additionally we have to check access permissions per index:

- payment attempts/intents => PaymentRead | PaymentWrite
- refunds => RefundRead | RefundWrite
- disputes => DisputeRead | DisputeWrite

### Additional Changes





## Motivation and Context


Checking access permissions to an index for a user is necessary to show results appropriately.

## How did you test it?


- Open the dashboard locally and hit the get_global_search_results API through the global search feature (command+k).
- Search using a query (eg: USD), to get the results corresponding to respective indexes (if user has access permission)
- Evaluate and verify the results

![Screenshot 2024-06-24 at 2 09 09 PM](https://github.com/juspay/hyperswitch/assets/83278309/9a86540a-4e76-44db-981f-175f1520b59e)

<img width="1683" alt="Screenshot 2024-06-25 at 11 26 25 PM" src="https://github.com/juspay/hyperswitch/assets/83278309/4a50c14f-59ca-48e0-ab10-81a5208445ba">

![Screenshot 2024-06-24 at 2 09 42 PM](https://github.com/juspay/hyperswitch/assets/83278309/44ec721a-2c38-4703-b0d2-442ac85121a7)

<img width="1706" alt="Screenshot 2024-06-25 at 11 26 44 PM" src="https://github.com/juspay/hyperswitch/assets/83278309/5123df9f-89bb-4433-85ab-b25988dd317c">


Alternatively, it can also be verified through Postman:
- Hit the /search and /search/{domain} API's to retrieve the data
<img width="1243" alt="Screenshot 2024-06-24 at 1 45 52 PM" src="https://github.com/juspay/hyperswitch/assets/83278309/b00a552d-128a-4fc1-bef2-a449aa6eef0c">
<img width="1234" alt="Screenshot 2024-06-24 at 2 16 30 PM" src="https://github.com/juspay/hyperswitch/assets/83278309/ef741c5f-d27f-435d-a86e-3c47c980b112">

## Checklist
